### PR TITLE
Add identifier examples

### PR DESCRIPTION
**Pull request by OR13**
This PR addresses:

- https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/issues/155

Related issues:

- https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/issues/154

### CODE REVIEW

**Comment by OR13** [comment]
Some additional details on the construction and use can be found here: https://github.com/transmute-industries/cose/blob/47e85916346af8560f3559ac28e98ad796dd0d34/src/scitt/identifiers/scitt-identifiers.test.ts

I'm not including these in the PR, because I don't understand the security properties required by "SCITT Identifiers", but if the code I wrote covers the security properties, I am happy to do a follow up PR, after this one is merged.

File: draft-ietf-scitt-architecture.md [hunk 1]
```diff
@@ -1091,3 +1096,99 @@ To indicate that the content is an scitt configuration represented as JSON:
 - Change Controller: IETF
 - Provisional registration?  No
 --- back
+
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:1099]:
```suggestion
```

File: draft-ietf-scitt-architecture.md [hunk 2]
```diff
@@ -1091,3 +1096,94 @@ To indicate that the content is an scitt configuration represented as JSON:
...
+# Identifiers
+
+This section provides informative examples of identifiers for statements, signed statements, and receipts.
+
+SCITT Identifiers are primarily meant to be understood by humans and secondarily meant to be understood by machines, as such we define text encodings for message identifiers first, and then provide binary translations according to standard transformations for URLs and URNs to binary formats. TODO add normative reference for how this is done.
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:1104]:
```suggestion
SCITT Identifiers are primarily meant to be understood by humans and secondarily meant to be understood by machines, as such we define text encodings for message identifiers first, and then provide binary translations according to standard transformations for URLs and URNs to binary formats.
TODO add normative reference for how this is done.
```

File: draft-ietf-scitt-architecture.md [hunk 3]
```diff
@@ -1091,3 +1096,94 @@ To indicate that the content is an scitt configuration represented as JSON:
...
+
+For each scitt conceptual message, we define a Data URL format according to {{-DataURLs}}, a URN format according to {{-URNs}} and a URL format according to {{URLs}}.
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:1106]:
```suggestion
For each SCITT conceptual message, we define a Data URL format according to {{-DataURLs}}, a URN format according to {{-URNs}} and a URL format according to {{URLs}}.
```

**Review by SteveLasker — APPROVED** [review]
LGTM

I pushed several markdown and spelling fixes

**Review by SteveLasker — APPROVED** [review]
LGTM for the consistency of CBOR structures

File: draft-ietf-scitt-architecture.md [hunk 4]
```diff
@@ -1095,3 +1103,96 @@ To indicate that the content is an scitt configuration represented as JSON:
...
+### Statement URL
+
+~~~http
+https://transparency.example/api/identifiers/urn:ietf:params:scitt:statement:sha-256:base64url:5i6UeRzg1...qnGmr1o
+~~~
+{: #example-statement-url align="left" title="Example Statement URL"}
+
+### Statement Data URL
+
+~~~http
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:1149]:
```suggestion
~~~
```

File: draft-ietf-scitt-architecture.md [hunk 5]
```diff
@@ -1095,3 +1106,194 @@ To indicate that the content is an scitt configuration represented as JSON:
...
+Let the `message-type`, be "signed-statement" for Signed Statements, and "receipt" for Receipts.
+
+~~~
+urn:ietf:params:scitt:\
+{message-type}:\
+{hash-name}:{base-encoding}:\
+{base64url-encoded-to-be-signed-bytes-digest}
+~~~
+
+Note that this means the content of the signature is not included in the identifer, even though signature related claims, such as activation or expiration information in protected headers are included.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:1180]: @fournet @ad-l per our call today, please make sure these are the security properties you want for "identifiers" for cose sign 1 messages.

File: draft-ietf-scitt-architecture.md [hunk 6]
```diff
@@ -1095,3 +1106,194 @@ To indicate that the content is an scitt configuration represented as JSON:
...
+~~~
+urn:ietf:params:scitt:\
+{message-type}:\
+{hash-name}:{base-encoding}:\
+{base64url-encoded-bytes-digest}
+~~~
+
+Note that because this identifier is computed over the unprotected header of the Signed Statement, any changes to the unprotected header, such as changing the order of the unprotected header map key value pairs, adding additional receipts, or adding additional proofs to a receipt, will change the identifier of a transparent statement.
+
+Note that because this identifier is computed over the signatures of the signed statement and signatures in each receipt, any cannonicalization of the signatures after the fact will produce a distinct identifier.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:1197]: @fournet @ad-l if you have use cases for this, please suggest them here.

If you have security concerns with this, please also leave suggestions.

Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:1197]: As there are open suggestions anymore and the feedback from @fournet was addressed by @OR13, I am moving on with approval and merge.

**Review by henkbirkholz — APPROVED** [review]

**Pull request merged**